internal/sessionmanager/sessionmanager.go

```diff
@@ -88,17 +99,31 @@ func (sm *SessionManager) NewSession(ctx context.Context,
 
 func (sm *SessionManager) Shutdown() {
 	sm.sessLk.Lock()
-	defer sm.sessLk.Unlock()
 
+	sessions := make([]Session, 0, len(sm.sessions))
 	for _, ses := range sm.sessions {
+		sessions = append(sessions, ses)
+	}
+
```

Comment:
nil out `sm.sessions` first?

Reply:
What does that do?

Reply:
I'm concerned we'll end up trying to shut sessions down twice without ever noticing that we have a bug. If we nil this out, we'll avoid that.

Reply:
I see makes sense 👍